toolchain/check/handle_index.cpp

```diff
@@ -20,60 +20,60 @@ auto HandleIndexExpressionStart(Context& /*context*/,
 // the array or tuple size. Returns the index on success, or nullptr on failure.
 static auto ValidateIntegerLiteralBound(Context& context,
                                         Parse::Node parse_node,
-                                        SemIR::Node operand_node,
-                                        SemIR::IntegerLiteral index_node,
+                                        SemIR::Inst operand_inst,
+                                        SemIR::IntegerLiteral index_inst,
                                         int size) -> const llvm::APInt* {
-  const auto& index_val = context.integers().Get(index_node.integer_id);
+  const auto& index_val = context.integers().Get(index_inst.integer_id);
   if (index_val.uge(size)) {
     CARBON_DIAGNOSTIC(IndexOutOfBounds, Error,
                       "Index `{0}` is past the end of `{1}`.", llvm::APSInt,
                       std::string);
     context.emitter().Emit(
         parse_node, IndexOutOfBounds,
         llvm::APSInt(index_val, /*isUnsigned=*/true),
-        context.sem_ir().StringifyType(operand_node.type_id()));
+        context.sem_ir().StringifyType(operand_inst.type_id()));
     return nullptr;
   }
   return &index_val;
 }
 
 auto HandleIndexExpression(Context& context, Parse::Node parse_node) -> bool {
-  auto index_node_id = context.node_stack().PopExpression();
-  auto index_node = context.nodes().Get(index_node_id);
-  auto operand_node_id = context.node_stack().PopExpression();
-  operand_node_id =
-      ConvertToValueOrReferenceExpression(context, operand_node_id);
-  auto operand_node = context.nodes().Get(operand_node_id);
-  auto operand_type_id = operand_node.type_id();
-  auto operand_type_node = context.nodes().Get(
+  SemIR::InstId index_inst_id = context.node_stack().PopExpression();
+  auto index_inst = context.insts().Get(index_inst_id);
+  SemIR::InstId operand_inst_id = context.node_stack().PopExpression();
```

Comment:
As discussed a bit on Discord, let's leave the `auto` usage consistent before and after here, and if we want to adjust, do that in a separate PR.

```suggestion
  auto index_inst_id = context.node_stack().PopExpression();
  auto index_inst = context.insts().Get(index_inst_id);
  auto operand_inst_id = context.node_stack().PopExpression();
```